dask/dataframe/core.py

```diff
@@ -3080,20 +3080,22 @@ def idxmaxmin_chunk(x, fn, axis=0, skipna=True, **kwargs):
     value = getattr(x, minmax)(axis=axis, skipna=skipna)
     n = len(x)
     if isinstance(idx, pd.Series):
-        return pd.DataFrame({'idx': idx, 'value': value, 'n': [n] * len(idx)}).T
-    return pd.DataFrame({'idx': [idx], 'value': [value], 'n': [n]}).T
+        chunk = pd.DataFrame({'idx': idx, 'value': value, 'n': [n] * len(idx)})
+        chunk['idx'] = chunk['idx'].astype(type(idx.iloc[0]))
+    else:
+        chunk = pd.DataFrame({'idx': [idx], 'value': [value], 'n': [n]})
+        chunk['idx'] = chunk['idx'].astype(type(idx))
+    print(chunk)
+    return chunk.T
```

Comment:
@mrocklin I tried explicitly setting the type here, which if you look at the printed values is correct.  But the concated chunks are all converted to floats.  Seems to be an issue in the concatenation itself.  I'll take a look, but noting it here.